llvm/lib/IR/Metadata.cpp

```diff
@@ -1590,7 +1589,7 @@ void Instruction::dropUnknownNonDebugMetadata(ArrayRef<unsigned> KnownIDs) {
   if (!Value::hasMetadata())
     return; // Nothing to remove!
 
-  SmallSet<unsigned, 4> KnownSet;
+  SmallDenseSet<unsigned, 32> KnownSet;
```

Comment:
Is SmallDenseSet here better than SmallSet? SmallSet performs linear scan in the small regime, while SmallDenseSet always uses a hash table.

Reply:
I measured the performance with rdtsc.  `SmallSet<unsigned, 32>` is about 1% faster than `SmallDenseSet<unsigned, 32>`.  I'm going back to `SmallSet`.